### ISSUE TITLE
Segmentation fault (core dumped)

### ISSUE THREAD

**Issue by buckshuang** [MIC-DKFZ/nnUNet#3051]
nnUNet_compile=False nnUNetv2_train Dataset001_PSMA 2d 0

############################
INFO: You are using the old nnU-Net default plans. We have updated our recommendations. Please consider using those instead! Read more here: https://github.com/MIC-DKFZ/nnUNet/blob/master/documentation/resenc_presets.md
############################

Using device: cuda:0

#######################################################################
Please cite the following paper when using nnU-Net:
Isensee, F., Jaeger, P. F., Kohl, S. A., Petersen, J., & Maier-Hein, K. H. (2021). nnU-Net: a self-configuring method for deep learning-based biomedical image segmentation. Nature methods, 18(2), 203-211.
#######################################################################

2026-07-07 10:52:49.310042: do_dummy_2d_data_aug: False
2026-07-07 10:52:49.310569: Creating new 5-fold cross-validation split...
2026-07-07 10:52:49.311875: Desired fold for training: 0
2026-07-07 10:52:49.311925: This split has 80 training and 20 validation cases.
using pin_memory on device 0
using pin_memory on device 0

This is the configuration used by this training:
Configuration name: 2d
 {'data_identifier': 'nnUNetPlans_2d', 'preprocessor_name': 'DefaultPreprocessor', 'batch_size': 59, 'patch_size': [288, 192], 'median_image_size_in_voxels': [263.0, 172.0], 'spacing': [3.2699999809265137, 4.0728302001953125], 'normalization_schemes': ['CTNormalization', 'ZScoreNormalization'], 'use_mask_for_norm': [False, False], 'resampling_fn_data': 'resample_data_or_seg_to_shape', 'resampling_fn_seg': 'resample_data_or_seg_to_shape', 'resampling_fn_data_kwargs': {'is_seg': False, 'order': 3, 'order_z': 0, 'force_separate_z': None}, 'resampling_fn_seg_kwargs': {'is_seg': True, 'order': 1, 'order_z': 0, 'force_separate_z': None}, 'resampling_fn_probabilities': 'resample_data_or_seg_to_shape', 'resampling_fn_probabilities_kwargs': {'is_seg': False, 'order': 1, 'order_z': 0, 'force_separate_z': None}, 'architecture': {'network_class_name': 'dynamic_network_architectures.architectures.unet.PlainConvUNet', 'arch_kwargs': {'n_stages': 6, 'features_per_stage': [32, 64, 128, 256, 512, 512], 'conv_op': 'torch.nn.modules.conv.Conv2d', 'kernel_sizes': [[3, 3], [3, 3], [3, 3], [3, 3], [3, 3], [3, 3]], 'strides': [[1, 1], [2, 2], [2, 2], [2, 2], [2, 2], [2, 2]], 'n_conv_per_stage': [2, 2, 2, 2, 2, 2], 'n_conv_per_stage_decoder': [2, 2, 2, 2, 2], 'conv_bias': True, 'norm_op': 'torch.nn.modules.instancenorm.InstanceNorm2d', 'norm_op_kwargs': {'eps': 1e-05, 'affine': True}, 'dropout_op': None, 'dropout_op_kwargs': None, 'nonlin': 'torch.nn.LeakyReLU', 'nonlin_kwargs': {'inplace': True}}, '_kw_requires_import': ['conv_op', 'norm_op', 'dropout_op', 'nonlin']}, 'batch_dice': True} 

These are the global plan.json settings:
 {'dataset_name': 'Dataset001_PSMA', 'plans_name': 'nnUNetPlans', 'original_median_spacing_after_transp': [4.0728302001953125, 3.2699999809265137, 4.0728302001953125], 'original_median_shape_after_transp': [200, 299, 200], 'image_reader_writer': 'SimpleITKIO', 'transpose_forward': [1, 0, 2], 'transpose_backward': [1, 0, 2], 'experiment_planner_used': 'ExperimentPlanner', 'label_manager': 'LabelManager', 'foreground_intensity_properties_per_channel': {'0': {'max': 3624.55908203125, 'mean': 205.56787109375, 'median': 132.35411071777344, 'min': -1502.844970703125, 'percentile_00_5': -1024.018310546875, 'percentile_99_5': 1148.55419921875, 'std': 344.6829833984375}, '1': {'max': 226.64405822753906, 'mean': 7.446566581726074, 'median': 5.415867805480957, 'min': 0.0, 'percentile_00_5': 0.0, 'percentile_99_5': 46.09492492675781, 'std': 7.546541213989258}}} 

Segmentation fault (core dumped)

I got this segementation fault. Please help me to solve this problem.

nnunetv2: Version: 2.8.1
PyTorch and CUDA versions: 2.3.1+cu121 12.1
NVIDIA-SMI 530.30.02    Driver Version: 530.30.02    CUDA Version: 12.1

**Comment by buckshuang** [comment]
nnunetv2: Version: 2.8.1
PyTorch and CUDA versions: 2.3.1+cu121 12.1
NVIDIA-SMI 530.30.02 Driver Version: 530.30.02 CUDA Version: 12.1

I tried disabling multiprocessing, reduced batch size and changed to one channel input, but got the same segmentation fault.

**Comment by moritzlangenberg** [comment]
Hi @buckshuang,

thank you for your detailed report. Looking at the log, everything that gets printed happens before the crash. That last block ("global plan.json settings") is the final thing nnU-Net prints before it touches the GPU, and the segfault hits on the first CUDA operation right after it. A crash there, with no Python traceback, very likely sits below nnUNet, e.g. in the PyTorch/CUDA layer or a clashing library, rather than in nnUNet code. I ran your exact network config here and it trains fine, which lines up with that.

Some things would help me finding the root cause:
1. which GPU is this? Can paste the full nvidia-smi output, not just the driver line? Your driver (530.30.02) is from early 2023, and if the card is newer than that driver/torch really supports, a segfault on the first kernel is exactly what you'd see.
2. Can you check if something like this is working?: `python -X faulthandler -c "import torch; print(torch.__version__, torch.version.cuda, torch.cuda.get_device_name(0)); a = torch.rand(4, 2, 288, 192, device='cuda'); w = torch.rand(32, 2, 3, 3, device='cuda'); b = torch.nn.functional.conv2d(a, w, padding=1); torch.cuda.synchronize(); print('bare CUDA conv OK', b.shape)"`. If that already segfaults, it is something in PyTorch/CUDA/driver combo and not nnUNet. Then reinstalling torch to match your driver, or updating the driver can help. If it prints "bare CUDA conv OK", then it's more likely a library conflict and the next step will show us where.

Third, can you try to get an actual backtrace so we can see which library is faulting? Can you try `nnUNet_n_proc_DA=0 nnUNet_compile=False gdb -batch -ex run -ex bt --args python -X faulthandler -m nnunetv2.run.run_training Dataset001_PSMA 2d 0` ?. Running "dmesg | tail" right after the crash often names the .so involved too. Whatever the backtrace points at will tell us whether this is a driver/torch mismatch or a conflict between libraries in your environment.

Best,
Moritz
